pkg/controller/rpaasinstance/rpaasinstance_controller.go

```diff
@@ -249,7 +264,31 @@ func (r *ReconcileRpaasInstance) updateLocationValues(instance *v1alpha1.RpaasIn
 
 		location.Value = value
 	}
+	return nil
+}
+
+func (r *ReconcileRpaasInstance) listConfigs(instance *v1alpha1.RpaasInstance) (*corev1.ConfigMapList, error) {
+	configList := &corev1.ConfigMapList{}
+	listOptions := &client.ListOptions{Namespace: instance.ObjectMeta.Namespace}
+	labelSelector := fmt.Sprintf("instance=%s,type=config", instance.Name)
+
+	if err := listOptions.SetLabelSelector(labelSelector); err != nil {
+		logrus.Errorf("Failed to query nginx configs: %v", err)
+		return nil, err
+	}
 
+	err := r.client.List(context.TODO(), listOptions, configList)
```

Comment:
Why not just add an inline matching labels? This way you can drop the lines at 272-279.

```golang
listOptions := client.MatchingLabels{"instance": instance.Name, "type": "config"}.
   ApplyToList(&client.ListOptions{Namespace: instance.Namespace})
```